src/app/shared/components/toolbar/toolbar.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core'
+import { MediaObserver } from '@angular/flex-layout'
+import { AuthService } from '@modules/core/auth/auth.service'
+import { combineLatest } from 'rxjs'
+import { tap } from 'rxjs/operators'
+import { SubSink } from 'subsink'
+
+@Component({
+  selector: 'cav-toolbar',
+  templateUrl: './toolbar.component.html',
+  styles: [
+    `
+      .image-cropper {
+        width: 40px;
+        height: 40px;
+        position: relative;
+        overflow: hidden;
+        border-radius: 50%;
+        margin-top: -8px;
+      }
+    `,
+  ],
+})
+export class ToolbarComponent implements OnInit, OnDestroy {
+  private subs = new SubSink()
+  title = 'RentMe'
+  opened = false
+
+  @Output() toggleSideNav = new EventEmitter<void>()
+
+  constructor(public authService: AuthService, public media: MediaObserver) {}
+
+  ngOnInit(): void {
+    this.subs.sink = combineLatest([
```

Comment:
Non sarebbe meglio inserire tutta questa logica in un metodo e chiamare quel metodo nel ngOnInit ?

Reply:
A che pro?

Reply:
Se il metodo avesse un nome esplicativo, in futuro basterà leggere solo il nome del metodo per capire cosa fa senza il bisogno di riguardarsi ogni volta tutta la la logica

Reply:
Questa porzione di codice non viene riutilizzata da nessuno ed estrarla in funzione in questo caso non ha senso.

Poi, se hai capito quello che fa questa porzione di codice, tu che nome consiglieresti che sia piu' esplicativo di quello che fa questa funzione nelle sue 4 linee di codice?

Uno deve imparare a capire quando estrarre del codice in una funzione ha senso e quando no, in caso non ha senso.